### PR TITLE
Remove broken optional native dependency

### PR DESCRIPTION
**Pull request by Gudahtt**
All native dependencies that fail to compile when installing have been removed from the project.

The only remaining example of this was a transitive dependency of `dnode`, which was an unused direct dependency. It has been removed. It was not imported anywhere, but it was referenced as a variable name in one unused module. This module has also been removed.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -5499,22 +5467,7 @@ eth-json-rpc-middleware@^6.0.0:
     pify "^3.0.0"
     safe-event-emitter "^1.0.1"
 
-eth-keyring-controller@^6.1.0:
```
Comment (Gudahtt) [inline yarn.lock:5502]: This was brought in by an outdated version of `@metamask/controllers` used by the swaps controller, so it wasn't actually used in practice. So this should have no impact whatsoever.

Comment (Gudahtt) [reply yarn.lock]: It looks like this update was also included in #2865 , so it's no longer present in this PR.

**Review by rickycodes — APPROVED** [review]
just tested these changes locally and they're looking good to me.

thanks @Gudahtt !

**Comment by Gudahtt** [comment]
One more lockfile update was required after merging in `develop`, as the lockfile was made invalid in a way git didn't detect as a conflict. That's what the 8cda96c commit is.

**Pull request merged**